### ISSUE TITLE
test findfirst and keys

### ISSUE THREAD

**Issue by dlfivefifty** [JuliaApproximation/InfiniteArrays.jl#8]

**Comment by goretkin** [comment]
Hi! Just FYI, some of this `findfirst` functionality is getting tested in `ApproxFun`, but is actually a little broken still: https://github.com/JuliaLang/julia/issues/36963

**Comment by dlfivefifty** [comment]
Yes, I made this issue a long time ago but it must have been that it was tested in ApproxFun.jl and wanted to add tests here (but didn't cause I was in a rush)
